logger/wandb_logger.py

```diff
@@ -0,0 +1,180 @@
+                            "domain": "pixel",
+                        }
+                    )
+            if boxes_data:
+                boxes = {
+                    "prediction": {
+                        "box_data": boxes_data,
+                        "class_labels": self.class_names,
+                    }
+                }
+            else:
+                boxes = None
+        if masks is not None:
+            final_mask = np.zeros(
+                (self.image.shape[0], self.image.shape[1]), dtype=np.uint8
+            )
+            for i, mask in enumerate(masks):
+                pred_mask = mask.mask
+                pred_class = int(classes[i]) + 1
+                final_mask = np.ma.array(final_mask, mask=pred_mask)
+                final_mask = final_mask.filled(pred_class)
+            final_mask = final_mask.astype(np.uint8)
+            masks = {
+                "prediction": {
+                    "mask_data": final_mask,
+                    "class_labels": self.class_names,
+                }
+            }
+        return {
+            "data_or_path": self.image_path,
+            "boxes": boxes,
+            "masks": masks,
+            "classes": self.class_set,
+        }
+
+
+class WandbInferenceLogger:
+    """
+    Logs inference images and predictions to wandb.
+    Currently, supports bounding boxes and instance segmentation.
+    """
+
+    def __init__(
+        self,
+        wandb_entity: str = None,
+        wandb_project: str = None,
+        run_name: str = None,
+        class_names: Dict[int, str] = None,
+        conf_threshold: float = 0.7,
+        config=None,
+    ):
+        if not is_wandb_available():
+            raise ImportError("Please install it using 'pip install wandb'.")
+
+        self.class_names = class_names if class_names else coco_label_map
+        self.wandb = wandb
+        self.run = None
+        if wandb.run:
+            self.run = wandb.run
+        else:
+            if wandb_project is None:
+                raise ValueError("wandb_project is required for wandb logger ")
+            self.run = wandb.init(
+                project=wandb_project,
+                name=run_name,
+                entity=wandb_entity,
+                config=config,
+            )
+        self.dataset_name = self.run.id + "_dataset"
+        self.conf_threshold = conf_threshold
+        self.table: wandb.Table = self.wandb.Table(columns=["file_id", "image"])
+
+    def log_inference(self, image, result):
+        """adds the inference result to a table in wandb."""
+        if not self.run:
+            return None
+        formatter = WandbFormatter(image, class_names=self.class_names)
+        image_name = str(Path(image).stem)
+        instance_prediction = formatter.convert_instance_predictions(
+            result["instances"]
+        )
+        self.table.add_data(image_name, self.wandb.Image(**instance_prediction))
```

Comment:
Is it possible to also log the number of bboxes in the prediction as well as the confidences, as it Soumik's v7 PR?